Rwaskiewicz/clean up intro (#1049)
ionic-team/stencil-site: docs/introduction/03-getting-started.md

- `npm run build` creates a production-ready version of your components. The components generated in this step are not
meant to be used in the local development server, but rather within a project that consumes your components.

- `npm test` runs your project's tests. The `create-stencil` CLI has created both end-to-end and unit tests when scaffolding your project.

At this time, Stencil does not interact with any version control systems (VCS) when running the `create-stencil` CLI.
If you wish to place your project under version control, we recommend initializing your VCS now.
If you wish to use git, run the following after changing your current directory to the root of your Stencil project:

```bash
$ git init
$ git add -A
$ git commit -m "initialize project using stencil cli" 
```

## My First Component

Stencil components are created by adding a new file with a `.tsx` extension, such as `my-component.tsx`.
The `.tsx` extension is required since Stencil components are built using [JSX](../components/templating-and-jsx.md) and TypeScript.

> `create-stencil` generated a component, `my-component.tsx` that can be found in the `src/components/my-component` directory.

Reviewer: ```suggestion
When we ran `create-stencil` above, it generated a component, `my-component.tsx`, that can be found in the `src/components/my-component` directory:
```

```tsx title="my-component.tsx"
import { Component, Prop, h } from '@stencil/core';
import { format } from '../../utils/utils';

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true,
})
export class MyComponent {
  @Prop() first: string;
  @Prop() middle: string;
  @Prop() last: string;

  private getText(): string {
    return format(this.first, this.middle, this.last);
  }

  render() {
    return <div>Hello, World! I'm {this.getText()}</div>;
  }
}
```

Once compiled, this component can be used in HTML just like any other tag.

```markup
<my-component first="Stencil" middle="'Don't call me a framework'" last="JS"></my-component>
```

When rendered, the browser will display `Hello World! I'm Stencil 'Don't call me a framework' JS`.

### Anatomy of `my-component`

Let's dive in and describe what's happening in `my-component`, line-by-line.

> After import statements, first piece we see is the [`@Component` decorator](../components/component.md):

Reviewer: ```suggestion
After the import statements, the first piece we see is the [`@Component` decorator](../components/component.md):
```

```tsx
@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true,
})
```
This decorator provides metadata about our component to the Stencil compiler.
Information, such as the custom element name (`tag`) to use, can be set here.
This decorator tells Stencil to:
- Set the [element's name](../components/component.md#tag) to 'my-component'
- [Apply the stylesheet](../components/component.md#styleurl) 'my-component.css' to the component
- Enable [native Shadow DOM functionality](../components/component.md#shadow) for this component

Below the `@Component()` decorator, we have a standard JavaScript class declaration:

```tsx
export class MyComponent {
```

Within this class is where you'll write the bulk of your code to bring your Stencil component to life.

Next, the component contains three class members, `first`, `middle` and `last`.
Each of these class members have the [`@Prop()` decorator](../properties#the-prop-decorator-prop) applied to them:
```ts
  @Prop() first: string;
  @Prop() middle: string;
  @Prop() last: string;
```

> `@Prop()` tells Stencil that the property is public to the component, and allows Stencil to update what's rendered when any of these public properties change.

Reviewer: ```suggestion
`@Prop()` tells Stencil that the property is public to the component, and allows Stencil to rerender when any of these public properties change.
```

```tsx
  private getText(): string {
    return format(this.first, this.middle, this.last);
  }
  
  render() {
    return <div>Hello, World! I'm {this.getText()}</div>;
  }
```

This component's `render()` returns a `<div>` element, containing text to render to the screen.

The `render()` function uses all three class members decorated with `@Prop()`, through the `getText` function.
Declaring private functions like `getText` helps pull logic out of the `render()` function's JSX.

Any property decorated with `@Prop()` is also automatically watched for changes.
If a user of our component were to change the element's `first`, `middle`, or `last` properties, our component would fire its `render()` function again, updating the displayed content.

## Component Generator

The `create-stencil` CLI can generate new components for you.
If you used one of the starters, you can simply run the `generate` npm script in your project, which will start the interactive generator.

```shell npm2yarn
npm run generate
```

Or, you can invoke the Stencil CLI directly with the `generate` command (`g` for short).
If you don't have `stencil` installed globally, prefix the command with `npx`.

```shell

> stencil generate
> ```

Reviewer: possibly we could do

````suggestion
stencil generate
# or
stencil g
```
````

just to like fully illustrate what's mentioned?

> You can optionally pass the component tag name directly to the command.
> The component tag name needs to be lowercase and contain at least one dash ('-').

Reviewer: Suggest adding a little example, just to make it unambiguous what we're saying

````suggestion
You can optionally pass the component tag name directly to the command.
The component tag name needs to be lowercase and contain at least one dash ('-').

```sh npm2yarn
stencil generate my-new-component
```
````

The generator will ask you which files to generate. 
This allows you to bootstrap a stylesheet as well as spec and e2e tests along with the component file.

All components will be generated within the `src/components` folder.
Within that directory, a folder will be created with the same name as the component tag name you provided, and within that folder the files will be generated.
It is also possible to specify one or multiple sub-folders to generate the component in.

For example, if you specify `pages/page-home` as the component tag name, the files will be generated in `src/components/pages/page-home`.

```shell
stencil generate pages/page-home
```

> ```plain
> src
>  |- components
>      |- pages
>          |- page-home
>              |- page-home.css
>              |- page-home.e2e.ts
>              |- page-home.spec.ts
>              |- page-home.tsx
> ```

Reviewer: Made a dummy directory and then ran `tree` on it to get slightly nicer symbols:

````suggestion
```plain
src
└── components
    └── pages
        └── page-home
            ├── page-home.css
            ├── page-home.e2e.ts
            ├── page-home.spec.ts
            └── page-home.tsx
```
````

looks like this rendered in Docusaurus:

<img width="811" alt="Screenshot 2023-03-02 at 10 30 10 AM" src="https://user-images.githubusercontent.com/6207644/222473418-9969ed8b-7dea-406b-9a99-32774a8b4dbb.png">
